torch/distributed/elastic/rendezvous/api.py

```diff
@@ -35,6 +38,61 @@ class RendezvousGracefulExitError(RendezvousError):
     Exception is a mechanism to exit the stack, however does not mean a failure.
     """
 
+@dataclass
+class RendezvousStoreInfo:
+    """Store address and port that can be used to bootstrap trainer distributed comms"""
+    MASTER_ADDR_KEY: ClassVar[str] = "MASTER_ADDR"
+    MASTER_PORT_KEY: ClassVar[str] = "MASTER_PORT"
+    master_addr: str
+    master_port: int
+
+    @staticmethod
+    def build(rank: int, store: Store) -> "RendezvousStoreInfo":
+        """Builds a rendezvous store info by finding unused new port on rank0 host.
+
+        If master_addr/master_port is knowns (useful when sharing existing tcp store server) use the constructor.
+        """
+        # TODO swap to collectives comms API
+        if rank == 0:
+            addr = socket.getfqdn()
+            port = _get_free_port()
+            store.set(RendezvousStoreInfo.MASTER_ADDR_KEY, addr.encode(encoding="UTF-8"))  # type: ignore[arg-type]
+            store.set(RendezvousStoreInfo.MASTER_PORT_KEY, str(port).encode(encoding="UTF-8"))  # type: ignore[arg-type]
+
+        addr = store.get(RendezvousStoreInfo.MASTER_ADDR_KEY).decode(encoding="UTF-8")
+        port = int(store.get(RendezvousStoreInfo.MASTER_PORT_KEY).decode(encoding="UTF-8"))
+        return RendezvousStoreInfo(master_addr=addr, master_port=port)
+
+
+class RendezvousInfo:
+    """Holds the information about the rendezvous."""
+    def __init__(self, store: Store, rank: int, world_size: int, bootstrap_store_info: Optional[RendezvousStoreInfo] = None):
+        self._store = store
+        self._rank = rank
+        self._world_size = world_size
+        self._bootstrap_store_info = bootstrap_store_info or RendezvousStoreInfo.build(rank=self._rank, store=self._store)
```

Comment:
I don't like that we're hiding store operations in the constructor of this -- I think this should be a pure data class. If the rendezvous wants to do this operation it should explicitly call it and return it